### DISCUSSION TITLE
Problema HA su docker

### DISCUSSION THREAD

**Discussion by GrupponeMissionario** [indomus/forum#827] [GENERALE - Discussioni generiche]
Ciao a tutti
dopo un aggiornamento di HA installato su docker, HA non parte più.
L'aggiornamento l'ho fatto con i seguenti comandi:
`sudo docker stop home-assistant`
`sudo docker rm home-assistant`
`sudo docker pull ghcr.io/home-assistant/home-assistant:stable`
`sudo docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v /home/homeassistant/.homeassistant:/config --net=host --restart=always ghcr.io/home-assistant/home-assistant:stable`

Il log dopo l'aggiornamento restituisce questo:

`tail -f /home/homeassistant/.homeassistant/home-assistant.log
    self._hass.async_run_job(listener, event)`
`  File "/usr/src/homeassistant/homeassistant/core.py", line 452, in async_run_job`
`    return self.async_run_hass_job(HassJob(target), *args)`
`  File "/usr/src/homeassistant/homeassistant/core.py", line 436, in async_run_hass_job`
`    return self.async_add_hass_job(hassjob, *args)`
`  File "/usr/src/homeassistant/homeassistant/core.py", line 368, in async_add_hass_job`
`    task = self.loop.create_task(hassjob.target(*args))`
`TypeError: stop() takes 1 positional argument but 2 were given`
`2021-10-05 22:42:48 WARNING (MainThread) [haffmpeg.core] FFmpeg isn't running!`
`2021-10-05 22:42:55 ERROR (SyncWorker_7) [homeassistant.components.command_line]` `Command failed: cat/sys/class/thermal/thermal_zone0/temp
`

Potete aiutarmi?
Grazie.

**Comment by indomus** [comment]
In italiano, per favore :)

**Reply by GrupponeMissionario** [reply]
ok scusate :D

**Comment by snakuzzo** [comment]
quell'errore non penso inibisca l'avvio di HA ... o almeno dovrebbe essere avviato in safe mode.
Quel log non penso sia completo. Poi dovresti dirci dove hai installato HA...per caso lo stai usando su RPI3 ? In tal caso devi installare su RaspOS una versione nuova di libsecomp. Leggi [QUI](https://indomus.it/news/aggiornamento-2021-7-per-home-assistant-core/)

**Reply by GrupponeMissionario** [reply]
Ecco il log completo:
[Nuovo documento di testo.txt](https://github.com/indomus/forum/files/7295933/Nuovo.documento.di.testo.txt)

Può essere che ho fatto un salto di versioni troppo pretenzioso?
Ho un Raspi4

**Reply by snakuzzo** [reply]
In base al log che hai mandato mi pare strano che NON PARTA proprio HA.
Dovrebbe partire in safe mode.
Ad ogni modo al tuo posto guarderei tutti gli ERROR, riga per riga, partendo dall'alto e li sistemerei uno per uno.
Non so che passaggio hai fatto, ma se salti da una versione vecchia di mesi all'ultima, di sicuro potresti avere qualcosa che non va. A quel punto ti vai a leggere i vari changelog di Core e piano piano sistemi guardando appunto i messaggi nel log

**Reply by GrupponeMissionario** [reply]
credo di aver avuto la versione di gennaio 2021 prima... ad occhio... ho modo di risalirne esattamente in qualche modo?
Sul fatto che non parta, ho anche provato a monitorare con:
`sudo docker stats`
e si vede che il docker di HA parte, mantiene le risorse a zero, poi scompare, poi riparte, e così via.... mentre il docker di duckdns rimane sempre in lista e si vede che consuma RAM e CPU...

**Reply by snakuzzo** [reply]
Comunque vedo questo...

`2021-10-05 22:04:50 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from modemtim.homenet.telecomitalia.it (192.168.1.1) (Mozilla/5.0 (Linux; Tizen 5.5; SAMSUNG SM-R820) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.106 Mobile Safari/537.36)`

Prova da altro browser...e prova a vedere se il problema ce l'hai solo in locale o anche da remoto.
Disabilita il ban.
Verifica la configurazione di eventuale reverse proxy

**Reply by GrupponeMissionario** [reply]
Quelli erano tentativi di Login a vuoto dall'app per Galaxy watch... non mi ero accorto che il Token era scaduto. L'aggiornamento l'ho fatto dopo.
Adesso non arrivo proprio alla pagina di login.

**Reply by snakuzzo** [reply]
fai così...mettiti in tail sui log e prova.
Se non funziona e non logga niente, il problema non è su HA

**Reply by GrupponeMissionario** [reply]
Ho anche provato a creare un altro docker, e installare HA da zero... sembra abbia lo stesso problema... il docker parte, si ferma, parte, si ferma ecc.

Allora se mi metto in tail esce:
```
pi@raspberrypi:~ $ tail -f /home/homeassistant/.homeassistant/home-assistant.log
    self._hass.async_run_job(listener, event)
  File "/usr/src/homeassistant/homeassistant/core.py", line 452, in async_run_job
    return self.async_run_hass_job(HassJob(target), *args)
  File "/usr/src/homeassistant/homeassistant/core.py", line 436, in async_run_hass_job
    return self.async_add_hass_job(hassjob, *args)
  File "/usr/src/homeassistant/homeassistant/core.py", line 368, in async_add_hass_job
    task = self.loop.create_task(hassjob.target(*args))
TypeError: stop() takes 1 positional argument but 2 were given
2021-10-05 22:42:48 WARNING (MainThread) [haffmpeg.core] FFmpeg isn't running!
2021-10-05 22:42:55 ERROR (SyncWorker_7) [homeassistant.components.command_line] Command failed: cat/sys/class/thermal/thermal_zone0/temp
```
se su un altro terminale do prima:
```
pi@raspberrypi:~ $ sudo docker stop home-assistant
home-assistant

```
poi:
```
pi@raspberrypi:~ $ sudo docker start home-assistant
home-assistant
```
sul tail non cambia nulla.

**Reply by snakuzzo** [reply]
Ma se il container si riavvia di continuo, non devi guardare il log di HA...devi guardare cosa accade su docker...

`docker logs -f <nomecontainer>`

**Reply by GrupponeMissionario** [reply]
ok e come?

**Reply by snakuzzo** [reply]
Ti ho scritto il comando

**Reply by GrupponeMissionario** [reply]
scusami non l'ho visto... sono fuso
```
pi@raspberrypi:~ $ docker logs -f home-assistant
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Fatal Python error: init_interp_main: can't initialize time
Python runtime state: core initialized
PermissionError: [Errno 1] Operation not permitted

Current thread 0xb6f12020 (most recent call first):
<no Python frame>
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Fatal Python error: init_interp_main: can't initialize time
Python runtime state: core initialized
PermissionError: [Errno 1] Operation not permitted

Current thread 0xb6f3b020 (most recent call first):
<no Python frame>
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
pi@raspberrypi:~ $
```

**Reply by snakuzzo** [reply]
Perdonami, ma mi spieghi che senso ha postare 2000 righe di log ?
Non vedi che sono praticamente tutte uguali e si ripetono di continuo e che l'errore sta già nel primo blocco? ...

```
Fatal Python error: init_interp_main: can't initialize time
Python runtime state: core initialized
PermissionError: [Errno 1] Operation not permitted
```

Hai letto quello che ti ho detto di leggere [QUI](https://github.com/indomus/forum/discussions/827#discussioncomment-1435541) in merito a libsecomp ? PUoi trovare anche i dettagli sia nei commenti del nostro articolo, sia sul [changelog ufficiale della 2021.7](https://www.home-assistant.io/blog/2021/07/07/release-20217/)

Io ora non so che OS stai usando di preciso, ma sostanzialmennte se stai utilizzando un OS a 32bit su CPU ARM devi assicurarti di avere innanzi tutto una versione di Docker superiore alla 19.03.9 epoi installare la versione di libsecomp superiore o uguale alla 2.4.2

``` 
sudo apt-get upgrade docker
wget https://ftp.debian.org/debian/pool/main/libs/libseccomp/libseccomp2_2.5.1-1_armhf.deb
sudo dpkg -i libseccomp2_2.5.1-1_armhf.deb
```

Inoltre ti consiglierei anche di dare un'occhiata a [questo post](https://community.home-assistant.io/t/migration-to-2021-7-fails-fatal-python-error-init-interp-main-cant-initialize-time/320648/10?u=snakuzzo) sul forum ufficiale di HA

**Reply by GrupponeMissionario** [reply]
> ...per caso lo stai usando su RPI3 ? In tal caso devi installare su RaspOS una versione nuova di libsecomp. Leggi [QUI](https://indomus.it/news/aggiornamento-2021-7-per-home-assistant-core/)

scusa ma questa frase per me significa che devo installare una nuova versione di libsecomp solo se sto usando RPI3... io sto usando RPI4.

Qui L'OS:
```
pi@raspberrypi:~ $ cat /etc/*release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
Qui la versione di docker (l'avevo verificata prima dell'aggiornamento):
```
pi@raspberrypi:~ $ sudo docker version
Client: Docker Engine - Community
 Version:           20.10.9
 API version:       1.41
 Go version:        go1.16.8
 Git commit:        c2ea9bc
 Built:             Mon Oct  4 16:06:55 2021
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.9
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.8
  Git commit:       79ea9d3
  Built:            Mon Oct  4 16:04:47 2021
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.4.11
  GitCommit:        5b46e404f6b9f661a205e28d59c982d3634148f8
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

ok credo di aver installato libseccomp2 seguendo quanto mi hai detto...
adesso rifaccio la procedura di aggiornamento HA e vediamo che succede...

EDIT

già meglio... credo.. il container non ci chiude più, adesso rimane aperto...
```
pi@raspberrypi:~ $ sudo docker stats
CONTAINER ID   NAME             CPU %     MEM USAGE / LIMIT     MEM %     NET I/O          BLOCK I/O         PIDS
b132e364a903   home-assistant   0.06%     60.77MiB / 3.513GiB   1.69%     0B / 0B          7.86MB / 475kB    12
```

Però HA non parte, allego il log
[log.txt](https://github.com/indomus/forum/files/7297126/log.txt)

**Comment by GrupponeMissionario** [comment]


**Reply by snakuzzo** [reply]
1. Quando rispondi devi rispondere sotto al messaggio (come ho fatto io ora) e non creando un nuovo thread
2. Hai messo tutto il log su una riga...devi separare le righe di log (che cominciano tutte con una data). Probabilmente hai sbagliato ad usare il markdown. Quando copi un blocco di righe devi usare tre backtick prima e tre alla fine.

Ti inviterei a ripostare correttamente perché così è impossibile leggere il log.

Grazie
